Modules/Core/include/selxNetworkBuilder.hxx

```diff
@@ -388,6 +404,12 @@ NetworkBuilder< ComponentList >::GetSourceInterfaces()
   {
     ComponentBase::Pointer component = componentSelector.second->GetComponent();
 
+    if(component == nullptr) {
```

Comment:
Do we need these checks (in release mode)? This for debugging our own code only, not for the user. (Compiling in Debug mode exists for this purpose). We need to make sure that there is no logic in the networkbuilder that leads to having a nullptr here.

Reply:
Therefore it should also fail hard when we get a nullptr here right? Otherwise we might not know when it goes wrong. Right?

Reply:
Ok, it is a matter of taste. To what extent do we double-check our own functionality and message about it? The function GetComponent returns a nullptr deliberately if there is not only one component. Run time, this means it fails hard at this point, which we detect with the unit tests (do we? we should) and can debug easilly. Putting extra checks and messages that might hint what kind of bug we have in our code only complicates and pollutes the code. It is of no use to the user should not be part of the executable users run. The logic in the networkbuilder was meant that it never reaches this point. We should fix that and make unit tests green. I agree we don't have a clear policy on when to throw things (simmilar issues might exist at other places), I think Coert made clear sugestions on that.